forc-util/src/lib.rs

```diff
@@ -407,7 +407,10 @@ fn construct_window<'a>(
     *start_ix -= std::cmp::min(calculated_start_ix, *start_ix);
     *end_ix -= std::cmp::min(calculated_start_ix, *end_ix);
     start.line = lines_to_start_of_snippet;
-    &input[calculated_start_ix..calculated_end_ix]
+
+    // The length of input must be at least as big as *end_ix. This is required by the
+    // annotate_snippets library
+    &input[calculated_start_ix..std::cmp::max(calculated_start_ix + *end_ix, calculated_end_ix)]
```

Comment:
This whole function is hard to read.  Is `end_ix` an index or a length?  Adding two indices together doesn't make sense does it?  I can't tell what's going on.

Reply:
You know what? I'm going to study that whole function tomorrow and rewrite it. This change is a bandaid fix and will probably lead to more issues in the future.